File: src/main/scala/robosky/uplands/client/render/entity/model/HexahaenModel.scala
package robosky.uplands.client.render.entity.model

import com.mojang.blaze3d.platform.GlStateManager

import java.util.Collections

import net.minecraft.client.render.entity.model.CompositeEntityModel
import net.minecraft.client.model.ModelPart

import robosky.uplands.entity.HexahaenEntity

object HexahaenModel extends CompositeEntityModel[HexahaenEntity] {

  // must be set before creating cuboids
  this.textureWidth = 128
  this.textureHeight = 64

  private val model: ModelPart = new ModelPart(this, 0, 0)
  model.addCuboid(-10, 4, -10, 20, 20, 20)

  override def getParts: java.lang.Iterable[ModelPart] = Collections.singletonList(model)
}

File: src/main/java/robosky/uplands/client/render/entity/model/HexahaenModel.java
package robosky.uplands.client.render.entity.model;

import net.minecraft.client.model.ModelPart;
import net.minecraft.client.render.entity.model.CompositeEntityModel;
import robosky.uplands.entity.HexahaenEntity;

import java.util.Collections;

public class HexahaenModel extends CompositeEntityModel<HexahaenEntity> {
    private final ModelPart model;

    public HexahaenModel() {
        // width and height must be set before creating cuboids
        this.textureWidth = 128;
        this.textureHeight = 64;
        this.model = new ModelPart(this, 0, 0);
        model.addCuboid(-10, 4, -10, 20, 20, 20);
    }

    @Override
    public Iterable<ModelPart> getParts() {
        return Collections.singletonList(model);
    }

    @Override
    public void setAngles(HexahaenEntity entity, float f1, float f2, float f3, float f4, float f5) {
    }
}
